framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -214,7 +214,13 @@ public void updateInferredMethodParameterTypes(
             // https://github.com/typetools/checker-framework/issues/682
             return;
         }
-        ClassSymbol classSymbol = getEnclosingClassSymbol(receiverTree);
+        final MethodTree methodDefinitionTree = (MethodTree) atf.declarationFromElement(methodElt);
```

Comment:
declarationFromElement only returns a tree if the methodElt was declared in the current complication unit.
What about using `ElementUtils.enclosingClass(methodElt);` or `((MethodSymbol) methodElt).enclClass()` intstead?

Reply:
I just added a test case where the use of the method happens in a different compilation unit than the definition of the method, and the tests still pass. I think `declarationFromElement` will return non-null so long as the definition was provided as source, which is fine with me - I don't want to infer types for non-source files anyway.

Did you have a different example in mind than the test case that I just added?

Reply:
The test is in the same compilation unit, but a different class.(https://docs.oracle.com/javase/specs/jls/se8/html/jls-7.html#jls-7.3) Try adding a method in a different source file.  If the method was declared in a different compilation unit, `declarationFromElement` will return null.  I've clarified its documentation adce909822.

Reply:
Sorry, I forgot to actually push the new test: https://github.com/typetools/checker-framework/pull/2929/files#diff-77455f96d2150b3322a0e76c2df697a9R3

That test should fail if what you're saying about `declarationFromElement` is correct, because the method use is in a different compilation unit than the superclass with the method definition.

Reply:
Oh, I see for members of a class, the tree can always be found.  So from a correctness point of view, this code is fine.  
However, this code gets the tree only to immediately convert it back to the element in `getEnclosingClassSymbol`.  So why do you need the tree?

Reply:
I don't think I actually need the tree. The important thing is getting the declaration, and neither of the methods you suggested actually fetch the declaration.

Basically, I have a method invocation, and I need to find the element representing the class it was declared in. `declarationFromUse` does that, but it requires going through the tree.

Reply:
Elements represent declarations, so `methodElement` is the method declaration.

Reply:
Right - I understand your point now. I've simplified the code to use `ElementUtils.enclosingClass`